### DISCUSSION TITLE
[Process] Command not generating any output via Process, but same command at CLI works

### DISCUSSION THREAD

**Discussion by simonhamp** [symfony/symfony#44873] [Q&A] [answered]
I have a command with a `commandLine` that ends up looking like this:

```bash
'git' '--git-dir' '/path/to/project/.git' '--work-tree' '/path/to/project' 'diff' '-r' '-p' '-m' '-M' '--full-index' '--unified=12' '--staged' '--' 'composer.lock'
```

If I omit the final two arguments (the `--` and `composer.lock`), the command generates output which I can retrieve.

If I copy-paste the whole command as above into a terminal, I get the expected output.

However, when I run it via `Process` it doesn't return any output. I've been debugging in the depths of pipes and streams to find out why but can't find anything helpful.

Has anyone got any clues?

**Comment by helmis123** [comment]
Can u include just enough code to allow reproducing the problem ?

**Comment by simonhamp** [comment, marked as answer]
@helmis123 Here's some sample code:

```php
// This doesn't produce any output for composer.lock...
$process = new Process(['git', '--git-dir', '/path/to/project/.git', '--work-tree', '/path/to/project', 'diff', '-r', '-p', '-m', '-M', '--full-index', '--unified=3', '--staged', '--', 'composer.lock']);
$process->run();
$output = $process->getOutput();

// This does, including composer.lock...
$process = new Process(['git', '--git-dir', '/path/to/project/.git', '--work-tree', '/path/to/project', 'diff', '-r', '-p', '-m', '-M', '--full-index', '--unified=3', '--staged']);
$process->run();
$output = $process->getOutput();
```

However, note that both work exactly as intended when run directly from the command line. So it's quite baffling that they don't quite work the same when called via `Process`.

That said, **I did stumble upon a 'fix' (which hints at the cause of the issue)**. It's something to do with how paths are being resolved. So somehow when passed as a parameter via `Process`, `'composer.lock'` isn't being treated in the same way as it is via the CLI.

It seems that when using `Process` it's best to use fully qualified paths.

Here's what finally worked:

```php
$process = new Process(['git', '--git-dir', '/path/to/project/.git', '--work-tree', '/path/to/project', 'diff', '-r', '-p', '-m', '-M', '--full-index', '--unified=3', '--staged', '--', '/path/to/project/composer.lock']);
$process->run();
$output = $process->getOutput();
```
